src/hotspot/share/services/diagnosticCommand.hpp

```diff
@@ -727,9 +731,36 @@ class CompilerDirectivesAddDCmd : public DCmdWithParser {
   virtual void execute(DCmdSource source, TRAPS);
 };
 
-class CompilerDirectivesClearDCmd : public DCmd {
+class CompilerDirectivesReplaceDCmd : public DCmdWithParser {
```

Comment:
'replace' is 'clear'+'add' (we don't have indexes and don't mention top element). Additional command is made to avoid double refreshing when '-r' is provided. It is more robust as we don't know when methods affected 'clear' are finally compiled.